aiida_vasp/calcs/tests/test_vasp2w90.py

```diff
@@ -0,0 +1,155 @@
+
+
+@ONLY_ONE_CALC
+def test_write_chgcar(fresh_aiida_env, vasp2w90_calc_and_ref, vasp_chgcar):
+    """Test that CHGCAR file is written correctly"""
+    vasp_calc, _ = vasp2w90_calc_and_ref
+    chgcar, ref_chgcar = vasp_chgcar
+    vasp_calc.use_charge_density(chgcar)
+    inp = vasp_calc.get_inputs_dict()
+    with tempfile.NamedTemporaryFile() as temp_file:
+        vasp_calc.write_chgcar(inp, temp_file.name)
+        with open(temp_file.name, 'r') as result_chgcar_fo:
+            assert result_chgcar_fo.read() == ref_chgcar
+
+
+@ONLY_ONE_CALC
+def test_write_wavecar(fresh_aiida_env, vasp2w90_calc_and_ref, vasp_wavecar):
+    """Test that WAVECAR file is written correctly"""
+    vasp_calc, _ = vasp2w90_calc_and_ref
+    wavecar, ref_wavecar = vasp_wavecar
+    vasp_calc.use_wavefunctions(wavecar)
+    inp = vasp_calc.get_inputs_dict()
+    with tempfile.NamedTemporaryFile() as temp_file:
+        vasp_calc.write_wavecar(inp, temp_file.name)
+        with open(temp_file.name, 'r') as result_wavecar_fo:
+            assert result_wavecar_fo.read() == ref_wavecar
+
+
+# pylint: disable=protected-access
+def test_prepare(vasp_nscf_and_ref):
+    """Check that preparing creates all necessary files"""
+    vasp_calc, _ = vasp_nscf_and_ref
+    inp = vasp_calc.get_inputs_dict()
+    with SandboxFolder() as sandbox_f:
+        calc_info = vasp_calc._prepare_for_submission(sandbox_f, inp)
+        inputs = sandbox_f.get_content_list()
+    assert set(inputs) == {'INCAR', 'KPOINTS', 'POSCAR', 'POTCAR', 'CHGCAR', 'WAVECAR'}
+    assert 'EIGENVAL' in calc_info.retrieve_list
+    assert 'DOSCAR' in calc_info.retrieve_list
+    assert ('wannier90*', '.', 0) in calc_info.retrieve_list
+
+    vasp_calc.inp.parameters.update_dict({'icharg': 2})
+    inp = vasp_calc.get_inputs_dict()
+    with SandboxFolder() as sandbox_f:
+        calc_info = vasp_calc._prepare_for_submission(sandbox_f, inp)
+        inputs = sandbox_f.get_content_list()
+    assert set(inputs) == {'INCAR', 'KPOINTS', 'POSCAR', 'POTCAR', 'WAVECAR'}
+
+
+@ONLY_ONE_CALC
+def test_parse_with_retrieved(vasp_nscf_and_ref, ref_retrieved_nscf):
+    """Check that parsing is successful and creates the right output links"""
+    vasp_calc, _ = vasp_nscf_and_ref
+    parser = vasp_calc.get_parserclass()(vasp_calc)
+    success, outputs = parser.parse_with_retrieved({'retrieved': ref_retrieved_nscf})
+    outputs = dict(outputs)
+    assert success
+    assert 'bands' in outputs
+    assert 'dos' in outputs
+    assert 'results' in outputs
+
+
+def test_verify_success(vasp2w90_calc_and_ref):
+    """Check that correct inputs are successfully verified"""
+    vasp_calc, _ = vasp2w90_calc_and_ref
+    inp = vasp_calc.get_inputs_dict()
+    vasp_calc.verify_inputs(inp)
+
+
+def test_verify_fail(vasp2w90_calc_and_ref):
+    """Check that incorrect inputs are not verified"""
+    vasp_calc, _ = vasp2w90_calc_and_ref
+    inp = vasp_calc.get_inputs_dict()
+    inp.pop('kpoints')
```

Comment:
This is effectively retesting the superclass, please replace the bad input with something specific to vasp2wanier90 calcs.